### ISSUE TITLE
15$: Improve error on invalid JS expression in JSX <style>

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#883]
In the following code, `{` is interpreted as a JS escape. `background-color: red;` is not a valid JavaScript expression, so the code has a syntax error:
```javascript
<style>
  body {
    background-color: red;
  }
</style>
```
We should provide a nice diagnostic, like:
> '{' begins JavaScript code; enclose your <style> in '{\`' and '\`}' instead

**Comment by strager** [comment]
We should probably detect this error using a lexer transaction. If we are in a `<style>` tag, and we see `{`, begin a transaction. Parse a few tokens. If it looks like CSS which is invalid JavaScript (e.g. we see a `;`, or we see identifier+minus+identifier+colon), report a special diagnostic.
